### PR TITLE
feat(plugin-command-patching): add path option to patch command

### PR DESCRIPTION
**Pull request by roseline124**
this pr is about #5293.

**Background**

There are cases where I want to access the existing patched project folder if I patch the package again. However, the temporary link given by 'pnpm-cli' is complicated and difficult to access again.

**as is**

`pnpm patch <packageName>`

```
> pnpm patch @types/jest                     

You can now edit the following folder: /private/var/folders/hl/s51yjywn6hj4v0w_mhrv8l540000gn/T/828d303a78782154638821e9da7d97b5/user

```

**to be**

`pnpm patch <packageName> [--path | -p <directoryPath>]`

It makes developer to access patch package directory again.

```
> pnpm patch @types/jest -p ~/pnpm-patches/project/@types/jest

You can now edit the following folder: /Users/.../pnpm-patches/project/@types/jest
```

**how about this..?**

It would be better the patch directory prefix is set with `pnpm-patches/<project-name>` when given path option?

### CODE REVIEW

File: packages/plugin-commands-patching/src/patch.ts [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
 
 export function help () {
   return renderHelp({
     description: 'Prepare a package for patching',
-    descriptionLists: [],
+    descriptionLists: [{
+      title: 'Options',
+      list: [
+        {
+          description: 'set a package directory path for patching',
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/patch.ts]: I think this option name is confusing. Maybe something like `--extract-dir` would be better.

The description is also not clear to me. Also it should be capitalized.

File: packages/plugin-commands-patching/src/patch.ts [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+> &
+CreateStoreControllerOptions & {
+  path?: string
   reporter?: (logObj: LogBase) => void
 }
 
+/**
+ * @TODO
+ * 3. document 업데이트
+ */
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/patch.ts]: update or remove the comment

File: packages/plugin-commands-patching/src/patch.ts [hunk 3]
```diff
@@ -41,13 +69,16 @@ export async function handler (opts: PatchCommandOptions, params: string[]) {
...
   })
   const filesResponse = await pkgResponse.files!()
-  const tempDir = tempy.directory()
-  const userChangesDir = path.join(tempDir, 'user')
+
+  const patchPackageDir = createPackageDirectory(opts.path)
+  const userChangesDir = path.join(patchPackageDir, 'user')
   await Promise.all([
-    store.ctrl.importPackage(path.join(tempDir, 'source'), {
+    store.ctrl.importPackage(path.join(patchPackageDir, 'source'), {
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/patch.ts]: maybe the source dir should remain in the temp directory and only the user dir should be placed to the directory specified via the option.

File: packages/plugin-commands-patching/src/patch.ts [hunk 4]
```diff
@@ -58,3 +89,16 @@ export async function handler (opts: PatchCommandOptions, params: string[]) {
...
   return `You can now edit the following folder: ${userChangesDir}`
 }
+
+function createPackageDirectory (userDir?: string) {
+  if (!userDir) {
+    return tempy.directory()
+  }
+
+  if (fs.existsSync(userDir)) {
+    throw new Error(`The package directory already exists: '${userDir}'`)
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/patch.ts]: You should use the PnpmError class from `@pnpm/error`

File: packages/plugin-commands-patching/test/patch.test.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ test('patch and commit', async () => {
     registries: { default: `http://localhost:${REGISTRY_MOCK_PORT}/` },
     storeDir,
     userConfig: {},
+    path: './temp/user/is-positive',
```
Comment (zkochan) [inline packages/plugin-commands-patching/test/patch.test.ts]: add a new test.

**Review by zkochan — CHANGES_REQUESTED** [review]
Add changesets, as described in the contributing.md

File: packages/plugin-commands-patching/src/patch.ts (hunk 1 above)
Comment (roseline124) [reply packages/plugin-commands-patching/src/patch.ts]: I get it. How about 'user-dir' or 'patch-dir'?

Comment (zkochan) [reply packages/plugin-commands-patching/src/patch.ts]: no, maybe `--edit-dir` or `--modify-dir`

Comment (roseline124) [reply packages/plugin-commands-patching/src/patch.ts]: I changed the name to '--edit-dir'

**Review by zkochan — APPROVED** [review]

**Pull request merged**